src/handles/handleDiagnostic.ts

```diff
@@ -55,27 +56,133 @@ function getSecurity(
   return security !== undefined ? security : 1
 }
 
+function handleLinterRegex(output: string, config: ILinterConfig): Diagnostic[] {
+  const {
+    formatLines = 1,
+    formatPattern,
+    offsetLine = 0,
+    offsetColumn = 0,
+    securities = {},
+    sourceName
+  } = config
+  let diagnostics: Diagnostic[] = [];
+
+  if (!formatLines || !formatPattern) {
+    throw new Error('missing formatLines or formatPattern')
+  }
+
+  const { line, column, endLine, endColumn, message, security } = formatPattern[1]
+  const endLineLookup = endLine != null ? endLine : line
+  const endColumnLookup = endColumn != null ? endColumn : column
+  const lines = output.split('\n')
+
+  let str: string = lines.shift()
+  while(lines.length > 0 || str !== undefined) {
+    str = [str].concat(lines.slice(0, formatLines - 1)).join('\n')
+    const m = str.match(new RegExp(formatPattern[0]))
+    if (m) {
+      let diagnostic: Diagnostic = {
+        severity: getSecurity(securities[m[security]]),
+        range: {
+          start: {
+            // line and character is base zero so need -1
+            line: sumNum(m[line], -1, offsetLine),
+            character: sumNum(m[column], -1, offsetColumn)
+          },
+          end: {
+            line: sumNum(m[endLineLookup], -1, offsetLine),
+            character: sumNum(m[endColumnLookup], -1, offsetColumn)
+          }
+        },
+        message: formatMessage(message, m),
+        source: sourceName
+      };
+      diagnostics.push(diagnostic);
+    }
+    str = lines.shift()
+  }
+
+  return diagnostics
+}
+
+const variableFinder = /\$\{[^}]+}/g;
+
+function formatStringWithObject<T extends Record<string, any> | any[]>(
```

Comment:
If you're up for it, this can also be used when formatting regular expression matches as well:

```jsonc
"message": [4, " [", 5, "]" ]
// could be written as
"message": "${4} [${5}]"
```

It would be a breaking change.